test/jdk/javax/accessibility/4422535/SetCurrentAccessibleValueTest.java

```diff
@@ -0,0 +1,321 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 4422535
+ * @summary setCurrentAccessibleValue returns true only for an Integer
+ * @run main SetCurrentAccessibleValueTest
+ */
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import javax.swing.JButton;
+import javax.swing.JInternalFrame;
+import javax.swing.JProgressBar;
+import javax.swing.JScrollBar;
+import javax.swing.JSlider;
+import javax.swing.JSplitPane;
+import javax.swing.SwingUtilities;
+
+public class SetCurrentAccessibleValueTest {
+
+    public static void doTest() {
+
+        JButton jButton = new JButton();
+        JInternalFrame iFrame = new JInternalFrame();
+        JSplitPane jSplitPane = new JSplitPane();
+        JScrollBar jScrollBar = new JScrollBar();
+        JProgressBar jProgressBar = new JProgressBar();
+        JSlider jSlider = new JSlider();
+
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(Integer.valueOf(5))) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept Integer");
+        }
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(Float.valueOf(5))) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept Float");
+        }
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(Double.valueOf(5))) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept Double");
+        }
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(Long.valueOf(5))) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept Long");
+        }
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(Short.valueOf((short) 123))) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept Short");
+        }
+
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(BigInteger.ONE)) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept BigInteger.ONE");
+        }
+
+        if (!jButton.getAccessibleContext().getAccessibleValue()
+            .setCurrentAccessibleValue(new BigDecimal(BigInteger.ONE))) {
+            throw new RuntimeException("JButton's Accessible Value"
+                + " implementation doesn't accept new "
+                + "BigDecimal(BigInteger.ONE)");
+        }
+
+        if (!jScrollBar.getAccessibleContext().getAccessibleValue()
```

Comment:
maybe you could write a loop to run each component?

Reply:
Done. Simplified test code with a loop over the JComponents being tested.